test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/kubernetes/pkg/features"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/e2e/nodefeature"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("SplitDisk", nodefeature.SplitDisk, func() {
+	f := framework.NewDefaultFramework("split-disk-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	pressureTimeout := 10 * time.Minute
+	expectedNodeCondition := v1.NodeDiskPressure
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.KubeletSeparateDiskGC)
+		if !hasSplitFileSystem(ctx) {
+			ginkgo.Skip("it doesn't have split filesystem")
+		}
```

Comment:
We probably should error in that case. If one is running the test with feature gate enabled then it should not silently skip the case.

Reply:
Good thinking! Otherwise, we wouldn't notice missing support or misconfiguration.

Reply:
It's true that these tests are supposed to be run on the environment having the split filesystem.
However if this is enabled by default, for clusters having the combined filesystem, these tests should be skipped.
So I think we should keep it skip.
wdyt?

Reply:
We ran into the same for swap. It's a tricky problem because there are some jobs that run all the node features and need to opt out of special node features. I think we probably should fail as @saschagrunert mentioned to protect us from loosing the test coverage, thought that will mean more churn in test-infra repo as we have to comb through all the jobs and make sure none accidentally run this test

Reply:
@haircommander Are you referring to this?

https://github.com/kubernetes/kubernetes/blob/ba80f252bf557417b1ff8c3a75a3d4a53c90a0bd/test/e2e_node/swap_test.go#L74-L76
https://github.com/kubernetes/kubernetes/pull/125027

Considering that, although failing would be safer, I'll keep it as it is not to break other tests (and checking all tests and fixing them would be a tough work)

Reply:
@saschagrunert @haircommander, should we skip or error?

I am chasing, as I would like to get this merged this week, if possible. :crossed_fingers:

Reply:
Let’s skip the test in that case. I’m happy with that.